modules/rostests/apitests/shell32/ItemIDList.cpp

```diff
@@ -11,11 +11,45 @@
 
 enum { DIRBIT = 1, FILEBIT = 2 };
 
+static BYTE GetPIDLType(LPCITEMIDLIST pidl)
+{
+    return pidl && pidl->mkid.cb >= 3 ? (pidl->mkid.abID[0] & 0x7F) : 0;
+}
+
+struct FS95
+{
+    WORD cb;
+    BYTE type;
+    BYTE unknown;
+    UINT size;
+    WORD date, time;
+    WORD att;
+    CHAR name[ANYSIZE_ARRAY];
+
+    static BOOL IsFS(LPCITEMIDLIST p)
+    {
+        return (p && p->mkid.cb > 2) ? (p->mkid.abID[0] & 0x70) == 0x30 : FALSE;
+    }
+    static FS95* Validate(LPCITEMIDLIST p)
+    {
+        C_ASSERT(FIELD_OFFSET(FS95, name) == 14);
+        return p && p->mkid.cb > FIELD_OFFSET(FS95, name) && IsFS(p) ? (FS95*)p : NULL;
+    }
+};
+
 static int FileStruct_Att(LPCITEMIDLIST pidl)
 {
-    return pidl && pidl->mkid.cb > 14 ? *((WORD*)((BYTE*)(pidl) + 12)) : (1 << 31); // See FileStruct in pidl.h
+    C_ASSERT(FIELD_OFFSET(FS95, att) == 12);
+    FS95 *p = FS95::Validate(pidl);
+    return p ? p->att : (UINT(1) << 31);
 }
 
+#define TEST_CLSID(pidl, type, offset, clsid) \
+    do { \
```

Comment:
you could remove one indentation level in this definition (here the `do {` serves a similar functionality as the opening `{` of a function)

Reply:
That would put the do on the same indent as the #define, that does not sound right?